fasta/seeker.go

```diff
@@ -0,0 +1,100 @@
+package fasta
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"io"
+	"log"
+	"os"
+)
+
+// Seeker enables random access of fasta sequences using a pre-computed index.
+type Seeker struct {
+	file readSeekCloser
+	idx  Index
+}
+
+// Close the Seeker.
+func (rs *Seeker) Close() error {
+	return rs.file.Close()
+}
+
+// TODO
+// readSeekCloser is a TEMPORARY github actions is updated as the io.ReadSeekCloser
+// was added to the builtin io package in 2020.
+type readSeekCloser interface {
+	io.Reader
+	io.Seeker
+	io.Closer
+}
+
+// NewSeeker opens a fasta file and an fai index file and enables
+// seek functionality so the entire fasta file does not need to be
+// present in memory.
+//
+// If you input an empty string for 'index', NewSeeker tries to find
+// the index file as 'fasta'.fai.
+func NewSeeker(fasta, index string) *Seeker {
+	sr := new(Seeker)
+	var err error
+
+	sr.file, err = os.Open(fasta)
+	exception.FatalOnErr(err)
+
+	if index == "" {
+		index = fasta + ".fai"
+	}
+	sr.idx = readIndex(index)
+	return sr
+}
+
+// SeekByName returns a portion of a fasta sequence identified by chromosome name.
+// Input start and end should be 0-based start-open end-closed.
+func SeekByName(sr *Seeker, chr string, start, end int) []dna.Base {
+	idx, ok := sr.idx.nameMap[chr]
+	if !ok {
+		log.Fatalf("ERROR: could not find sequence for fasta record '%s'\n", chr)
+	}
+	return seek(sr, sr.idx.chroms[idx], start, end)
+}
+
+// SeekByIndex returns a portion of a fasta sequence identified by chromosome index (order in file).
+// Input start and end should be 0-based start-open end-closed.
```

Comment:
To make this match other places in go and most of gonomics.  Can we make this start-closed, end-open?

Reply:
Ah, it actually is. I just screwed up the comment. Fixed.